### DISCUSSION TITLE
Android WebXR stopped working

### DISCUSSION THREAD

**Discussion by Immortall1961** [google/model-viewer#4665] [Q&A]
Hello in the last few days I have a problem with the fact that WebXR stopped working in my browser on android. When I click on the button to enter AR, the model viewer freezes and this message shows up in the console: 
Attempting to present in AR with WebXR...

When I then click on the next model and enter AR, the model loads, but in Scene Viewer... This is what appears in the console: 
Attempting to present in AR with WebXR...
11:18:43 AM
▶ Cannot present while a model is already presenting
11:18:43 AM
▶ Error while trying to present in AR with WebXR
11:18:43 AM
▶ InvalidStateError: Failed to execute 'requestSession' on 'XRSystem': There is already an active, immersive XRSession.
11:18:43 AM
▶ Error while trying to end WebXR AR session
11:18:43 AM
▶ TypeError: Cannot read properties of null (reading 'end')
11:18:43 AM
▶ Falling back to next ar-mode
11:18:43 AM
Attempting to present in AR with Scene Viewer...

Any idea what to do? Is there a bug in the phone/viewer/model viewer? The examples from https://modelviewer.dev/examples/augmentedreality/index.html don't work for me either, they also run in Scene Viewer.

Thanks for your help

EDIT: AR from this topic is not for me as well: 
https://github.com/google/model-viewer/discussions/4659

EDIT2: 
Behaviour on iOS is somehow laggy too.  Last week it was ok

**Comment by elalish** [comment]
This is outside of MV - looks like a Chrome bug (assuming your browser is Chrome?). iOS is a separate issue - I'm going to ignore that unless there's a separate issue opened for it with a lot more detail.

Also, can you tell use what device, browser, and URL you're using for your repro? How widespread and repeatable is this failure?

@bialpio This looks like a Chrome WebXR regression - can you repro?

**Reply by Immortall1961** [reply]
Sorry for long response time. I have Samsung s22 and chrome (but i tested Samsung browser as well)  just like niseXR. So its maybe some Samsung Bug?

**Reply by rebrob** [reply]
We began having the same issue with Android not displaying AR using webxr a few months ago. We lost a lot of the model-viewer features like annotations, etc...

**Reply by elalish** [reply]
Most commonly this is due to XR permissions being denied - that is remembered on the site until manually unset. If it's another problem, please file an issue with detailed repro steps.

**Reply by rebrob** [reply]
Thanks for that info. We will add instructions for our users.

**Comment by niseXR** [comment]
@Immortall1961 i have exactly the same experience with it.
I'm using an Samsung S22 and Chrome Browser. 
https://modelviewer.dev/examples/augmentedreality/index.html freeze also on my phone.
Do someone know a workaround? like another version of modelviewer?
thx for your help and update!

**Reply by Immortall1961** [reply]
Did you somehow make it work again?

**Reply by niseXR** [reply]
nope. the only work around is to put scene-viewer in front. But my pbr-textures dosn't look good there and i have issues there with opacity in my model. Really sad, because i really don't know what to do. Apple works still fine.

**Comment by bialpio** [comment]
> @bialpio This looks like a Chrome WebXR regression - can you repro?

I tried to repro on Pixel 8, w/ Chrome 121.0.6167.164 (64-bit) - I am able to enter AR normally (and change the models using the carousel from within a running AR session). Scene Viewer only kicks in when I try the specific Scene Viewer example that is further down on the page (and it also works correctly).

I do see weirdness when testing on Samsung Galaxy S8+ though - only the model's shadows get rendered (but I can enter AR and see the shadows rendered on the ground plane in the environment). See attached picture for how it looks like pre-entry into AR (there are some WebGL errors that may be relevant).

![mv-only-shadow](https://github.com/google/model-viewer/assets/6945808/2dfc12fc-c7a1-42dd-9006-35b562ebf6fa)

**Reply by elalish** [reply]
Lovely, that looks like yet another bug - probably a GPU driver bug? Since you have the repro, would you mind pasting `?model-viewer-debug-mode` at the end of your URL and copying the shader compiler error it spits out to the console? That should be enough for a Three.js/Chrome bug.

**Reply by bialpio** [reply]
Screenshot attached. I don't really know how to interpret this output, looks like a non-printable character?..
![model-viewer-debug](https://github.com/google/model-viewer/assets/6945808/d0718f66-e455-4694-ac90-2c5e9e116160)

**Reply by elalish** [reply]
Bizarre - something looks deeply broken in Samsung's GPU driver. Do you think you might be able to file it somewhere that it would get the attention it deserves?

**Comment by niseXR** [comment]
Hey,
Nope. All my previous work stopped working.
I do all my creations with the modelviewer.  Its super frustrating.

Https://Nise-xr.com/P3
Https://nise-xr.com/fare/index2.html

Only 2 of my models which are off.


www.nisexr.com<https://nisexr.com/>



The information contained in this e-mail is confidential and is intended only for the addressee. If you are not the intended addressee, any disclosure, copying or distribution by you is prohibited. Should you receive this email in error, please immediately notify as soon as possibly vis email or telephone. Internet communications are not always secure and therefor NISE XR does not accept legal responsibility for the contents of this message.



-------- Ursprüngliche Nachricht --------
Von: Immortall1961 ***@***.***>
Datum: 12.02.24 11:50 (GMT+01:00)
An: google/model-viewer ***@***.***>
Cc: Sebastian Wieland ***@***.***>, Comment ***@***.***>
Betreff: Re: [google/model-viewer] Android WebXR stopped working (Discussion #4665)


Did you somehow make it work again?
